test/java/org/opendatakit/briefcase/ui/export/components/ConfigurationPanelTest.java

```diff
@@ -68,20 +70,32 @@ public void choose_pem_file_button_is_swapped_for_a_clean_button_that_cleans_the
   }
 
   @Test
-  public void shows_error_and_clears_field_after_inserting_an_invalid_start_date() {
-    component.setEndDate(LocalDate.of(2018, 1, 1));
-    component.setStartDate(LocalDate.of(2020, 1, 1));
-    assertThat(component.errorDialog(500), is(GenericUIMatchers.visible()));
-    component.acceptErrorDialog();
-    assertThat(component.startDateField(), is(empty()));
+  public void cannot_insert_an_invalid_end_date() {
+    component.setStartDate(LocalDate.of(2017, 1, 30));
+    assertFalse(component.endDatePicker().isDateAllowed(LocalDate.of(2017, 1, 28)));
   }
 
   @Test
-  public void shows_error_and_clears_field_after_inserting_an_invalid_end_date() {
-    component.setStartDate(LocalDate.of(2020, 1, 1));
-    component.setEndDate(LocalDate.of(2018, 1, 1));
-    assertThat(component.errorDialog(500), is(GenericUIMatchers.visible()));
-    component.acceptErrorDialog();
-    assertThat(component.endDateField(), is(empty()));
+  public void empty_start_date_end_datepicker_is_disabled() {
+    component.startDatePicker().clear();
+    assertFalse(component.endDatePicker().isEnabled());
   }
-}
\ No newline at end of file
+
+  @Test
+  public void clear_end_date_when_new_start_date_is_after_end_date() {
+    component.setStartDate(LocalDate.of(2017,1,22));
+    component.setEndDate(LocalDate.of(2017,1,25));
+    component.setStartDate(LocalDate.of(2017,1,27));
+    assertEquals(component.endDatePicker().getDate(), null);
+  }
+
+  @Test
+  public void new_start_date_is_before_or_on_end_date_is_valid() {
+    component.setStartDate(LocalDate.of(2017,1,20));
+    component.setEndDate(LocalDate.of(2017,1,25));
+    component.setStartDate(LocalDate.of(2017,1,24));
+    assertEquals(component.endDatePicker().getDate(), LocalDate.of(2017,1,25));
```

Comment:
Please, change these `assertEquals` to `assertThat(actualValue, is(expectedValue))` like the on the rest of the test suite.

Reply:
Done